### ISSUE TITLE
Multithreading with PointNeighbors.jl and Polyester.jl

### ISSUE THREAD

**Issue by kaipartmann** [kaipartmann/Peridynamics.jl#110]
There are differences in performance (see https://github.com/kaipartmann/Peridynamics.jl/pull/107#issuecomment-2182414620) that seem to be related to combining `Polyester.@batch` and `Base.Threads`.

`PointNeighbors.jl` uses `Polyester.@batch` within the [`@threaded`](https://github.com/trixi-framework/PointNeighbors.jl/blob/591827bdd986630f802642ce4f11a459588e0e67/src/util.jl#L53-L55) macro.

With `threaded_nhs_update=true` and changing all `Threads.@threads` calls in `Peridynamics.jl`, the following differences occur for the benchmark in https://github.com/kaipartmann/Peridynamics.jl/pull/107#issuecomment-2182414620:
```julia
# `@threads :static`
#  49.821336 seconds (2.93 M allocations: 3.263 GiB, 0.13% gc time)

# `@threads`
#  27.514394 seconds (2.92 M allocations: 3.262 GiB, 0.39% gc time)

# `Polyester.@batch`
#  11.450976 seconds (2.34 M allocations: 3.199 GiB, 0.49% gc time)
```

_Originally posted by @kaipartmann in https://github.com/kaipartmann/Peridynamics.jl/issues/107#issuecomment-2182525712_

**Comment by kaipartmann** [comment]
@efaulhaber

**Comment by kaipartmann** [comment]
Closed with #153.
